checker/src/main/java/alluxio/checker/MapReduceIntegrationChecker.java

```diff
@@ -0,0 +1,315 @@
+   */
+  private Status generateReport() throws Exception {
+    // Reads all the part-r-* files in MapReduceOutPutFile folder
+    FileStatus[] outputFileStatus = mFileSystem.listStatus(mOutputFilePath,
+        path -> path.getName().startsWith(("part-")));
+
+    Map<Status, List<String>> resultMap = new HashMap<>();
+    for (int i = 0; i < outputFileStatus.length; i++) {
+      Path curOutputFilePath = outputFileStatus[i].getPath();
+      try (BufferedReader curOutputFileReader = new BufferedReader(
+          new InputStreamReader(mFileSystem.open(curOutputFilePath)))) {
+        String nextLine = "";
+        while ((nextLine = curOutputFileReader.readLine()) != null) {
+          int sep = nextLine.indexOf("\t");
+          Status curStatus = Status.valueOf(nextLine.substring(0, sep).trim());
+          String curAddresses = nextLine.substring(sep + 1).trim();
+          List<String> addresses = resultMap.getOrDefault(curStatus, new ArrayList<>());
+          addresses.add(curAddresses);
+          resultMap.put(curStatus, addresses);
+        }
+      }
+    }
+
+    File reportFile = new File("./MapReduceIntegrationReport.txt");
+    try (PrintWriter reportWriter = new PrintWriter(reportFile)) {
+      // Prints the current time to separate integration checker results
+      SimpleDateFormat df = new SimpleDateFormat("dd/MM/yy HH:mm:ss");
+      Date date = new Date();
+      reportWriter.printf("%n%n%n***** The integration checker ran at %s. *****%n%n",
+          df.format(date));
+      Status resultStatus = CheckerUtils.printNodesResults(resultMap, reportWriter);
+      switch (resultStatus) {
+        case FAIL_TO_FIND_CLASS:
+          reportWriter.println(FAIL_TO_FIND_CLASS_MESSAGE);
+          reportWriter.println(TEST_FAILED_MESSAGE);
+          break;
+        case FAIL_TO_FIND_FS:
+          reportWriter.println(FAIL_TO_FIND_FS_MESSAGE);
+          reportWriter.println(TEST_FAILED_MESSAGE);
+          break;
+        default:
+          reportWriter.println(TEST_PASSED_MESSAGE);
+      }
+      return resultStatus;
+    }
+  }
+
+  /**
+   * Implements MapReduce with Alluxio integration checker.
+   *
+   * @return 0 for success, 2 for unable to find Alluxio classes, 1 otherwise
+   */
+  private int run(String[] args) throws Exception {
+    Configuration conf = new Configuration();
+    String[] otherArgs =  new GenericOptionsParser(conf, args).getRemainingArgs();
+    conf.set(MRJobConfig.NUM_MAPS, otherArgs[0]);
+    createHdfsFilesystem(conf);
+
+    Job job = Job.getInstance(conf, "MapReduceIntegrationChecker");
+    job.setJarByClass(MapReduceIntegrationChecker.class);
+    job.setMapperClass(FooMapper.class);
+    job.setCombinerClass(FooReducer.class);
+    job.setReducerClass(FooReducer.class);
+    job.setOutputKeyClass(Text.class);
+    job.setOutputValueClass(Text.class);
+    job.setInputFormatClass(EmptyInputFormat.class);
+    FileOutputFormat.setOutputPath(job, mOutputFilePath);
+
+    try {
```

Comment:
in order to enforce the close of `FileSystem`, a better practice is to use `try-with-resource` idiom since `mFileSystem` is `Closeable`. 
Complicating `finally` is less ideal.

Reply:
I want it to clean the MapReduceOutputFile and then close the filesystem, so I do not use the try-with-resources.  The logic here is a little complicated.